### DISCUSSION TITLE
Avoid loading entire disk images into memory for use with v86

### DISCUSSION THREAD

**Discussion by hello-smile6** [DustinBrett/daedalOS#47] [Ideas]
v86 allows booting disk images asynchronously from a URL. If you use https://github.com/humphd/nohost , you could make it possible to use Ubuntu 16 easier.

**Comment by DustinBrett** [comment]
Thanks for this idea.

I was aware of that feature but for some reason had considered not doing it. I can't recall why so I will give it another look. As it is though the 2 files I host are small and anything dragged on will be hosted locally client side in the browser.

As for using nohost, I hadn't heard of this and am not sure of the connection to this v86 suggestion or it's help with Ubtuntu, could you clarify how I could use it for those use cases?

Thanks!

**Comment by hello-smile6** [comment]
> Thanks for this idea.
> 
> I was aware of that feature but for some reason had considered not doing it. I can't recall why so I will give it another look. As it is though the 2 files I host are small and anything dragged on will be hosted locally client side in the browser.
> 
> As for using nohost, I hadn't heard of this and am not sure of the connection to this v86 suggestion or it's help with Ubtuntu, could you clarify how I could use it for those use cases?
> 
> Thanks!

Nohost would simply allow loading disk images without loading the entire image, as well as opening local files in the iframe browser. Also, I am aware that those disk images are small. However, I mounted a directory and was unable to boot a disk image from that local directory.

**Comment by DustinBrett** [comment]
Ah ok I see what you mean with nohost. Thanks for the feedback about not being able to load a mounted disk image, I will look into that and recently had the same issue. But if I drag and dropped the image onto the desktop it usually loaded ok.

**Reply by hello-smile6** [reply]
Not being able to boot a local one is frustrating. Also, could you either fix Git with mounted directories or create a separate website that lets Git be used with local directories? I currently have no way to use Git on Chrome OS.

**Reply by DustinBrett** [reply]
Those do sound like bugs, I'd like to try and reproduce this and if need be move it back to Issues. Could you clarify what disk image you used when you did "_I mounted a directory and was unable to boot a disk image from that local directory._". I tried to do this with Damn Small Linux on a shared folder and it actually worked.

![image](https://user-images.githubusercontent.com/16656318/152924352-8bea26be-ffec-4499-979f-43cacd36976f.png)

As for Git, at one point I thought it worked and also thought I'd fixed it, but trying it now I am noticing it gets stuck at `Analyzing workdir` when cloning to a local directory. Is that the issue you have?

![image](https://user-images.githubusercontent.com/16656318/152925258-8152c4a7-0802-4567-b85f-4be7619694ac.png)

Thanks!

**Reply by hello-smile6** [reply]
That's probably my problem. Also, it corrupts IndexedDB while doing Git on a mounted directory.

**Reply by DustinBrett** [reply]
Ya I need to try and see what is going on there. Maybe some kind of error is happening silently because it makes a little progress but then stops.

**Reply by hello-smile6** [reply]
Maybe just add `debugger` to the beginning of every function.

**Reply by DustinBrett** [reply]
Thanks but if you'd like to look into it you can. I know how to debug and anything I've posted on here for issues is not trivial and for the most part these are FYI things.

**Reply by hello-smile6** [reply]
I was suggesting that because that's what I do whenever things won't run, even though there are probably better ways.

**Reply by DustinBrett** [reply]
Most of the things I am posting here now are on my wall of things I haven't had time to do. Very few of them are total mysteries to me and I know how to solve them, I just don't have time. The next step after I post all these things is to post a discussion to try and see what people think I should focus my time on. But thank you for the suggestions.

**Reply by hello-smile6** [reply]
Okay
